src/coreclr/vm/syncblk.h

```diff
@@ -894,7 +894,7 @@ class InteropSyncBlockInfo
         LIMITED_METHOD_CONTRACT;
```

Comment:
I am not sure `AllocTaggedMemory` should be in the DAC as we don't tend to have mutating APIs in the DAC. 

Also, I `GetTaggedMemory`,  `m_taggedMemory`, and `m_taggedAlloc` are not properly DACized and could be dangerous. The main problem is if this API gets somehow called, and someone tried to read the pointer, they'll read in the address space of the debugger. I see that this is supposed to be "scratch memory", so maybe it doesn't make sense to read into it, but given they are pointers. If it always just reads from `m_taggedAlloc` and having a copy on the debugger side and trying to remote read this through the DAC makes no sense, then sure. I just have a knee jerk reaction over undacized pointers being "we won't need this" landmines for people working in diagnostics.

Reply:
I'd suggest:
void* m_taggedMemory -> PTR_VOID m_taggedMemory
void* GetTaggedMemory() -> PTR_VOID GetTaggedMemory()

In a non-DAC build PTR_VOID is defined as void* so product build is unchanged. When compiled in the DAC PTR_VOID is a smart pointer that unambiguously refers to the debuggee address space. If you attempt to read memory through it (which this current change never does) the smart pointer would copy the appropriate memory from the debuggee.

The risk entailed by having AllocTaggedMemory() compiled into the DAC build pre-dated this change, but while we are staring at it and talking about it we may as well tidy it up. Wrapping #ifndef DACCESS_COMPILE/#endif around the method eliminates the possibility of DAC code ever invoking it.

Reply:
Thanks for the guidance and explanation of concern.